src/Symfony/Component/DependencyInjection/Tests/Loader/PhpFileLoaderTest.php

```diff
@@ -164,4 +164,15 @@ public function testDeprecatedWithoutPackageAndVersion()
         $loader = new PhpFileLoader($container = new ContainerBuilder(), new FileLocator());
         $loader->load($fixtures.'/config/deprecated_without_package_version.php');
     }
+
+    public function testNestedBundleConfigNotAllowed()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessageMatches('/^'.preg_quote('Could not resolve argument "Symfony\\Config\\AcmeConfig\\NestedConfig $config"', '/').'/');
```

Comment:
But these two lines just over `$container->compile();`

That way we can be extra sure that the `InvalidArgumentException` is not thrown anywhere else.

Reply:
I removed `$container->compile();` because there only `$container->load()` needed. But I think that `$this->expectException*` should stay on the top of test.

Reply:
I know the Symfony code base is not consistent on this. But I believe it is good to always keep the `expectException*` as close to the bottom as possible. 

Say that `new PhpFileLoader` threw an `InvalidArgumentException`. Then this test would still be green but a bug has been introduced.

Sure using `expectExceptionMessageMatches()` mitigates this issue a bit. But I still believe they should be at the bottom. The way I see it there is also no argument to put them at the top (like many many people do), all the other assertions are at the bottom of the method.

Reply:
> But I think that `$this->expectException*` should stay on the top of test.

why ? This allows such an exception to be accepted anywhere in the test, while we want it only in `load()`

Reply:
> I know the Symfony code base is not consistent on this. But I believe it is good to always keep the `expectException*` as close to the bottom as possible.

the main reason of the inconsistency is that most of the Symfony tests have been written in the era of `@expectedException` rather than `$this->expectException()`. When migrating between them, the equivalent approach was to place them at the top (as the annotation was essentially placing that before the start of the method). We haven't done the work of improving the placement in each existing test. But new tests should go the better way.

Reply:
You are right, I was initially very obsessed with maintaining a consistency with existing tests.